src/parser/mod.rs

```diff
@@ -116,6 +117,8 @@ parser!(varargs location -> 'b () =
     ()
 );
 
+// Parses the return type of function.
+// This may involve effects.
```

Comment:
```suggestion
```
This comment seems outdated seeing as the `function_return_type` parser is separate from the `effect_set` parser in `function_definition`